lib/client/socket.js

```diff
@@ -27,7 +27,25 @@ module.exports = ({ url, token, filters, config, identifyingMetadata }) => {
     },
     pathname : `/primus/${token}`,
   });
-  const io = new Socket(url);
+
+  const io = new Socket(url, {
```

Comment:
Would potentially add a brief comment here saying something like:
`// Will exponentially back-off from x secs to a maximum of 20 mins, and retry for a total period of around y mins`

Reply:
Actually added this comment and then removed it :) Pulling it back :)